actions/discover/src/discover.ts

```diff
@@ -1,21 +1,36 @@
 import * as core from '@actions/core'
-import { exec } from 'child_process'
+import { getExecOutput } from '@actions/exec'
 import { quote } from 'shell-quote'
 
 export async function run(): Promise<void> {
   try {
-    const parse = core.getBooleanInput('parse_images')
     const paths = quote([core.getInput('paths')])
     const targets = core.getInput('targets')
 
-    const flags = parse ? ['-ji'] : ['-j']
+    const flags = ['-j']
     if (targets.trim() !== '') {
       flags.push(...targets.split(' ').map(t => `-t ${t}`))
     }
     const command = ['ci', 'scan', ...flags, paths].filter(Boolean).join(' ')
 
     core.info(`Running command: ${command}`)
-    core.setOutput('json', await execCommand(command))
+    const { stdout } = await getExecOutput(command)
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const parsedData = JSON.parse(stdout)
+
+    const pathsArray = []
+
+    for (const key in parsedData) {
+      if (Object.prototype.hasOwnProperty.call(parsedData, key)) {
+        pathsArray.push(key)
+      }
+    }
+
+    // JSON of path mapping to its filtered list of targets that need to be run
+    core.setOutput('json', stdout)
+    // List of path that should be run
+    core.setOutput('paths', pathsArray)
```

Comment:
Can you help me understand why we need two distinct outputs here? It looks like `paths` is just a transformed version of `json`.

Reply:
so it is easirt to manage in `run.yml`
```
 - name: Discover Earthly files
        uses: input-output-hk/catalyst-ci/actions/discover@feat/targets-scanner
        id: discover
        with:
          targets: ${{ inputs.target }}
      - name: Check for empty output
        id: check
        run: |
          json=$(echo '${{ steps.discover.outputs.json }}' | jq -rc)
          paths=$(echo '${{ steps.discover.outputs.paths }}' | jq -rc)
          if [ "$output" == "null" ]; then
            echo "json=[]" >> $GITHUB_OUTPUT
            echo "paths=[]" >> $GITHUB_OUTPUT
          else
            echo "json=$json" >> $GITHUB_OUTPUT
            echo "paths=$paths" >> $GITHUB_OUTPUT
          fi
```

Reply:
To avoid calling the filtering target multiple times, I decided to retrieve the map of Earthfile paths and their corresponding filtered targets. Consequently, the output here needs to be modified from the previous version. Instead of `json` being a list of Earthfile paths, this list is now placed in `paths`. The `json` now represents all the discovered Earthfiles along with a list of filtered targets. This information will be used to extract the necessary targets to run during the matrix.